File: Assignment 4/C/02. height_weight.c
#include <stdio.h>

typedef struct data{
    int height;
    int weight;
}data;

void maxRatio(data arr[],int n){
    float max = arr[0].weight/arr[0].height;
    int idx = 0;
    for(int i=1; i<n; i++){
        float t = arr[i].weight/arr[i].height;
        if(t>max){
            max = t;
            idx = i;
        }
    }
    printf("%d has max ratio(w/h) = %f",idx+1,max);
}

int main(){
    int n;
    printf("enter no. of student: ");
    scanf("%d",&n);

    data arr[n];
    for(int i=0; i<n; i++){
        printf("enter height(cm) of %d: ",i+1);
        scanf("%d",&arr[i].height);
        printf("enter weight(kg) of %d: ",i+1);
        scanf("%d",&arr[i].weight);
    }

    maxRatio(arr,n);
    return 0;
}


//OUTPUT:

// enter no. of student: 2
// enter height(cm) of 1: 11
// enter weight(kg) of 1: 22
// enter height(cm) of 2: 33
// enter weight(kg) of 2: 33
// 1 has max ratio(w/h) = 2.000000

File: Assignment 4/Python/02. height_weight .py
class data:
    def __init__(self,h,w):
        self.h = h
        self.w = w

def maxRatio(arr):
    max_ratio = arr[0].w/arr[0].h
    idx = 0
    for i in range(1, len(arr)):
        current_ratio = arr[i].w / arr[i].h
        if current_ratio > max_ratio:
            max_ratio = current_ratio
            idx = i
    print(f"Student {idx + 1} has the max ratio (w/h) = {max_ratio}")

n = int(input("Enter the number of students: "))

arr = []
for i in range(n):
    height = int(input(f"Enter height (cm) of student {i + 1}: "))
    weight = int(input(f"Enter weight (kg) of student {i + 1}: "))
    arr.append(data(height, weight))

maxRatio(arr)


# OUTPUT:
# Enter the number of students: 3
# Enter height (cm) of student 1: 12
# Enter weight (kg) of student 1: 23
# Enter height (cm) of student 2: 2
# Enter weight (kg) of student 2: 45
# Enter height (cm) of student 3: 34
# Enter weight (kg) of student 3: 56
# Student 2 has the max ratio (w/h) = 22.5
